R/read_sc_all_home_care.R

```diff
@@ -0,0 +1,39 @@
+#' Read Social Care Home Care data
+#'
+#' @return an extract of the data as a [tibble][tibble::tibble-package].
+#'
+read_sc_all_home_care <- function() {
+
+  # Read in data---------------------------------------
+
+  # set-up conection to platform
+  db_connection <- phs_db_connection(dsn = "DVPROD")
+
+  # read in data - social care 2 home care
+  home_care_data <- dplyr::tbl(db_connection, dbplyr::in_schema("social_care_2", "homecare_snapshot")) %>%
```

Comment:
What if we passed the connection as a parameter? That way we'd only have to open the connection once and could use it for all of the SC functions.

Have set a default, so that if you don't pass a connection it will still work by asking you to create one.
```suggestion
read_sc_all_home_care <- function(sc_dvprod_connection = phs_db_connection(dsn = "DVPROD")) {

  # Read in data---------------------------------------

  # read in data - social care 2 home care
  home_care_data <- dplyr::tbl(sc_dvprod_connection, dbplyr::in_schema("social_care_2", "homecare_snapshot")) %>%
```

Reply:
I think this would be a good fix as this can also be implemented for other sc functions using the sc platform